fenwick/src/main/java/org/opencadc/fenwick/AllArtifacts.java

```diff
@@ -80,4 +84,15 @@
     public AllArtifacts() {
         super();
     }
+
+    /**
+     * Obtain the iterator of clauses used to build a query to include the Artifacts being merged.
+     *
+     * @return An empty iterator.
+     */
+    @Override
+    public Iterator<String> iterateIncludeClauses() {
+        log.debug("No filters for AllArtifacts.");
+        return Collections.emptyIterator();
```

Comment:
I think this needs to return an iterator with a single no-op condition... 0-length string could be made to work but could easily be interpretted as a mistake elsewhere. Otherwise, artifact sync loop is going to have to special-case the empty iterator to do a single query, rather than just do a query for each clause returned by the iterator.

Reply:
There is a change here, but returning the string "EOF" won't allow the sync loop to blindly pass that to the query generator. That should end up doing a query like this:
```
select * from inventory.Artifact where (EOF) and ...
```  
The goal is for a configuration with the AllArtifacts selector to execute a single query with no filtering on content. To do that without treating this selector as special, the iterator needs to return one item that can be included in the WHERE clause of the query without filtering out any artifacts. All these would work: `id is not null`, `uri is not null`, and 0-length (or whitespace) string, but I'm not sure which of those would be least/most confusing to a developer in the future. Ops people would never see it except if they look at the queries executed in luskan... then they might also get confused by some of them. 

If it was me and only I was ever going to look at the code, I'd just put a 0-length string right in the
```
return Collections.singletonList("").iterator();
```
and make sure that worked correctly... yeah, this is looking forward to implementing the sync loop and query generator, which will look something like:
```
Iterator<String> filters = selector.iterator();
while (filters.hasNext()) {
   String adql = generateADQL(filters.next());
   ResourceIterator<Artifact> ri = tapClient.execute(adql, artifactMapper);
   ...
}
```
To do that, AllArtifacts has to return one string that can be blindly put in the where clause of the ADQL query. 

**Other clarifications that would be good to add now:**

The ArtifactSelector javadoc for the iterator() method should specify exactly what is going to be done with each string. 

I would restrict the exceptions that are throwable from ArtifactSelector.iterator()... Maybe specify these standard exceptions for a few predictable cases - this is probably enough:
* ResourceNotFoundException when required configuration is missing
* IOException for unreadable configuration (eg impls can just let those fly)
* IllegalStateException for invalid configuration content
It's possible that if we were to allow for retrieving config from a URL that many/all the exceptions thrown by HttpTransfer.prepare() would be possible

Reply:
Thank you.  I thought there would be a special case for ignoring the SQL, such as EOF.